engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -880,6 +883,20 @@ private void execute() throws YarnException, IOException
         if (0 != exitStatus) {
           if (allocatedContainer != null) {
             numFailedContainers.incrementAndGet();
+            if (exitStatus != 1) {
+              // If container failure due to framework
+              AtomicInteger failedTimes = new AtomicInteger(1);
+              String hostname = allocatedContainer.container.getNodeId().getHost();
+              if (failedContainersMap.containsKey(hostname)) {
```

Comment:
ConcurrentMap is needed only if later down the line, we want to expose failure count per hostname as part of ClusterAppStats. Similar to getFailedContainers (returns total number of failures)